ppl/zqd/redis.go

```diff
@@ -0,0 +1,66 @@
+package zqd
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/go-redis/redis/v8"
+)
+
+func NewRedisClient(ctx context.Context, conf RedisConfig) (*redis.Client, error) {
+	if !conf.Enabled {
+		return nil, nil
+	}
+	client := redis.NewClient(&conf.Options)
+	if cmd := client.Ping(ctx); cmd.Err() != nil {
+		return nil, cmd.Err()
+	}
+	return client, nil
+}
+
+type RedisConfig struct {
+	redis.Options
+
+	Enabled bool
+}
+
+func (c *RedisConfig) SetFlags(fs *flag.FlagSet) {
+	fs.Var(c, "redis.url", "redis connection url (i.e. redis://<user>:<password>@<host>:<port>/<db_number>)")
+	fs.BoolVar(&c.Enabled, "redis.enabled", false, "enable use of a redis server")
+	fs.StringVar(&c.Addr, "redis.addr", "localhost:6379", "redis address")
+	fs.StringVar(&c.Username, "redis.user", "", "redis username")
+	fs.StringVar(&c.Password, "redis.password", "", "redis password")
```

Comment:
Would you please add a `PasswordFile` option, like you did with postgres?

Reply:
I would prefer adding this in the next pr when that option will actually be used